device/iot-device-samples/pnp-device-sample/temperature-controller-device-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/device/PnpConvention.java

```diff
@@ -92,18 +105,39 @@ public static Message createIotHubMessageUtf8(@NonNull String telemetryName, @No
      *     }
      * }
      */
-    public static Set<Property> createPropertyPatch(@NonNull final String propertyName, @NonNull final Object propertyValue, String componentName) {
+    public static Set<Property> createComponentPropertyPatch(@NonNull final String propertyName, @NonNull final Object propertyValue, @NonNull String componentName) {
+
+        return createComponentPropertyPatch(componentName, new HashMap<String, Object>()
+        {{
+            put(propertyName, propertyValue);
+        }});
+    }
+
```

Comment:
Note: no need to have CreatePropertyPatch taking hashmap since sendReportedProperties already takes a Set<Property> and is easy to directly use that.

Set<Property> reportProperties = new HashSet<Property>()
            {
                {
                    add(new Property("prop1", 70));
                    add(new Property("prop2", "value"));
                }
            };